### PR TITLE
Fixes Remove Dependant

### PR DESCRIPTION
**Pull request by FuriousLlama**
# Fixes or Implements [AB#9494](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/9494)

* [ ] Enhancement
* [x] Bug
* [ ] Documentation

## Description

Added missing logic

## Testing

* [ ] Unit Tests Updated
* [ ] Functional Tests Updated
* [x] Not Required

### Steps to Reproduce

Open the dependant's page, delete a dependant

### UI Changes

NO

### CODE REVIEW

**Review by leobcgov — DISMISSED** [review]
We though it was missing the User Delegate injection

File: Apps/Database/src/Delegates/DBUserDelegateDelegate.cs [hunk 1]
```diff
@@ -91,14 +91,9 @@ public DBResult<IEnumerable<UserDelegate>> Get(string delegateId, int page, int
         public DBResult<UserDelegate> Delete(string ownerId, string delegateId, bool commit)
         {
             this.logger.LogTrace($"Deleting UserDelegate (ownerId: {ownerId}, delegateId: {delegateId}) from DB...");
-            UserDelegate userDelegate = new UserDelegate()
```
Comment (sslaws) [inline Apps/Database/src/Delegates/DBUserDelegateDelegate.cs:94]: We should give this back to whomever did delete.  We should not be doing a query to do a delete, instead we should be passing in the object that was originally fetched or marginally the key identifiers along with the version number.

**Review by sslaws — CHANGES_REQUESTED** [review]

File: Apps/Database/src/Delegates/DBUserDelegateDelegate.cs (hunk 1 above)
Comment (FuriousLlama) [reply Apps/Database/src/Delegates/DBUserDelegateDelegate.cs:94]: at this point the whole fetch does not have a version in it, so....

Comment (sslaws) [reply Apps/Database/src/Delegates/DBUserDelegateDelegate.cs:94]: UserDelegate is defined as public class UserDelegate : AuditableEntity anything that is Auditable has a version attached to it.  The object returned by GetUserDelegate and InsertUserDelegate is returning the full object.  We should be using that...

Comment (FuriousLlama) [reply Apps/Database/src/Delegates/DBUserDelegateDelegate.cs:94]: I've updated the model to contain a representation of the user delegate to correctly use the concurrency checks

File: Apps/WebClient/src/Server/Models/DependentModel.cs [hunk 2]
```diff
@@ -26,29 +26,51 @@ public class DependentModel
...
-        /// Gets or sets the Masked PHN of the dependent.
+        /// Gets or sets the hdid which has delegated access to the owner Id.
+        public string DelegateId { get; set; } = null!;
+
+        /// <summary>
+        /// Gets or sets the version.
         /// </summary>
-        [JsonPropertyName("maskedPHN")]
-        public string MaskedPHN { get; set; } = string.Empty;
+        public uint Version { get; set; }
```
Comment (sslaws) [inline Apps/WebClient/src/Server/Models/DependentModel.cs:43]: No documentation on this property, is it not warning you on web client?

**Review by sslaws — CHANGES_REQUESTED** [review]
Please ensure no smells introduced across projects.

File: Apps/WebClient/src/Server/Models/DependentModel.cs (hunk 2 above)
Comment (FuriousLlama) [reply Apps/WebClient/src/Server/Models/DependentModel.cs:43]: Not sure what you mean. These are the docs I see 
```
        /// <summary>
        /// Gets or sets the version.
        /// </summary>
```

**Review by tiagograf-AOT — APPROVED** [review]

**Review by sslaws — APPROVED** [review]

**Pull request merged**